### PR TITLE
Handle error on reader.read error

### PR DESCRIPTION
**Pull request by joshualim92**
**Description**

Ran into an edge case where if the network is interrupted during a download, an uncaught network error occurs. Unfortunately FileLoader isn't able to recover from here because the url is still listed in the module level `loading` object. There is already a catch handler for the fetch call which should handle this case so this is just added in so the error handling is propagated up.

*This contribution is funded by [VRIFY](https://vrify.com)*

### CODE REVIEW

File: src/loaders/FileLoader.js [hunk 1]
```diff
@@ -122,30 +122,35 @@ class FileLoader extends Loader {
...
+											readData();
 
 										}
 
-										controller.enqueue( value );
-										readData();
+									} ).catch( err => {
 
-									}
+										throw err;
```
Comment (gkjohnson) [inline src/loaders/FileLoader.js:152]: What are you expecting this to do? Throwing the same error you caught is effectively a no op.

Comment (joshualim92) [reply src/loaders/FileLoader.js]: Apologies, when I was running this locally I thought this was sufficient but it was only because something went wrong with my hot reload.

I've replaced the commit I've had before so hopefully it'll be clearer in what the intent is. I'm open to altering the approach though. I'm not sure if a shared error handler is preferred here, another solution is to switch this to be an async/await so it's easier to have an outer error handle for the recursive promises.

File: src/loaders/FileLoader.js [hunk 2]
```diff
@@ -116,38 +116,40 @@ class FileLoader extends Loader {
...
-										controller.close();
+											loaded += value.byteLength;
 
-									} else {
+											const event = new ProgressEvent( 'progress', { lengthComputable, loaded, total } );
+											for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
 
-										loaded += value.byteLength;
+												const callback = callbacks[ i ];
+												if ( callback.onProgress ) callback.onProgress( event );
```
Comment (gkjohnson) [inline src/loaders/FileLoader.js:139]: Can we revert these indentation changes? It makes it hard to follow what has meaningfully changed. Please take make a minimal change we can more easily follow the code adjustments.

Comment (joshualim92) [reply src/loaders/FileLoader.js:139]: Updated

**Comment by gkjohnson** [comment]
Great thanks - can you provide steps on how to reproduce the issue for testing?

**Comment by joshualim92** [comment]
> Great thanks - can you provide steps on how to reproduce the issue for testing?

For sure, the steps involve using the dev tools to get since it's an edge case error while downloading. For ease of testing I created a code sandbox, please let me know if you'd prefer a different format

https://codesandbox.io/s/tender-sinoussi-cjnzko

For ease of testing it uses one of our models that's a little larger.

Steps:
1. Go to chrome dev tools and throttle the download speed (for ease of turning offline)
2. Once the .glb file starts downloading, switch the throttle to "Offline"

Expected:
onError callback should be called in `loader.load`

Actual:
Error isn't called back and instead is uncaught in promise

Screen Shot 2022-08-08 at 2 47 36 PM

**Comment by Mugen87** [comment]
Should be fixed via #28216.
